### PR TITLE
[Snyk] Security upgrade transformers from 4.53.0 to 5.5.0

### PR DESCRIPTION
**Pull request by karly-quick**
snyk-top-banner

#### Snyk changed the following file(s):

- `models/tt_transformers/requirements.txt`

#### Breaking Change Risk

Merge Risk: High

> **Notice:** This assessment is enhanced by AI.

---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.
> - Some vulnerabilities couldn't be fully fixed and so Snyk will still find them when the project is tested again. This may be because the vulnerability existed within more than one direct dependency, but not all of the affected dependencies could be upgraded.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: 
🧐 [View latest project report](https://app.snyk.io/org/tt-test-rust/project/1b5e0150-313e-44f7-b854-5865b605407b?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/tt-test-rust/project/1b5e0150-313e-44f7-b854-5865b605407b?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Learn about vulnerability in an interactive lesson of Snyk Learn.](https://learn.snyk.io/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"breakingChangeRiskLevel":"high","FF_showPullRequestBreakingChanges":true,"FF_showPullRequestBreakingChangesWebSearch":false,"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"transformers","from":"4.53.0","to":"5.5.0"}],"env":"prod","issuesToFix":["SNYK-PYTHON-TRANSFORMERS-17675094"],"prId":"c8d14863-3ea5-453d-9f8a-3f2f8724f6f7","prPublicId":"c8d14863-3ea5-453d-9f8a-3f2f8724f6f7","packageManager":"pip","priorityScoreList":[180],"projectPublicId":"1b5e0150-313e-44f7-b854-5865b605407b","projectUrl":"https://app.snyk.io/org/tt-test-rust/project/1b5e0150-313e-44f7-b854-5865b605407b?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["updated-fix-title","priorityScore"],"type":"auto","upgrade":[],"vulns":["SNYK-PYTHON-TRANSFORMERS-17675094"],"patch":[],"isBreakingChange":false,"remediationStrategy":"vuln"}'

### CODE REVIEW

**Comment by karly-quick** [comment]
Merge Risk: High

The upgrade from `transformers` v4 to v5 is a **HIGH** risk major version update with extensive and critical breaking changes. This is the first major version upgrade in five years and it modernizes the entire library, requiring significant developer action.  
  
**Key Breaking Changes:**  
  
*   **PyTorch-Only Framework:** Support for TensorFlow (all `TF*` model classes) and JAX/Flax (all `Flax*` model classes) has been completely **removed**. The library is now exclusively for PyTorch. Applications using other backends will fail and require a full migration to PyTorch.  
*   **Environment Requirements:** The minimum required Python version is now 3.10+, and PyTorch must be version 2.4 or higher.  
*   **Model Loading Behavior:** The `from_pretrained()` method now defaults to `dtype="auto"`, which loads models in their native precision (e.g., float16) instead of float32. This can silently affect model outputs and requires explicit setting of `torch_dtype=torch.float32` to retain the old behavior.  
*   **API & Parameter Removals:**  
    *   The `use_auth_token` parameter is removed everywhere and replaced by `token`.  
    *   Legacy quantization arguments (`load_in_4bit`, `load_in_8bit`) are removed in favor of a new `quantization_config` object.  
    *   Features like `BetterTransformer`, head masking, head pruning, and `torchscript` support have been removed.  
    *   The `transformers-cli` command has been replaced by `transformers serve`.  
  
Source: [Migration Guide](https://github.com/huggingface/transformers/blob/main/MIGRATION_GUIDE_V5.md), [Release Notes](https://huggingface.co/docs/transformers/v5.0.0/en/release_notes)  
  
**Recommendation:** This upgrade will almost certainly break existing applications. Do not merge without a thorough migration effort. Developers must carefully review the official migration guides and validate their models, data loading, and training/inference pipelines.

> **Notice 🤖:** This content was augmented using **artificial intelligence**. AI-generated content **may** contain **errors** and should be **reviewed** for accuracy before use.

[//]: # (snyk:breakability:{"publicId":"700046e8-4092-4b6b-b053-6e9c6d853ce9"})

**Comment by karly-quick** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/tt-test-rust/pr-checks/4af88b37-f513-4f86-be82-69886487a734?refs=cc92e1acd01263b6e9849ff26c0fc6d01eec1bdc&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/tt-test-rust/pr-checks/4af88b37-f513-4f86-be82-69886487a734/license?refs=cc92e1acd01263b6e9849ff26c0fc6d01eec1bdc&source=prChecksComment) |
| :white_check_mark: | Code Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/tt-test-rust/project/c45c259d-ddf0-4be1-960e-6f383f9dbef2/pr-check/571951d9-13d0-4b8a-a899-ac5de757a9a5?refs=cc92e1acd01263b6e9849ff26c0fc6d01eec1bdc&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).
